src/Interpreters/Context.h

```diff
@@ -18,6 +18,10 @@
 
 #include "config_core.h"
 
+#if !defined(ARCADIA_BUILD)
```

Comment:
btw no need for `#if !defined(ARCADIA_BUILD)` anymore.

Reply:
Ok.

Reply:
@kssenii I have some trouble when introducing `hive_metastore.thrift` to CH. Because compiling `hive_metastore.thrift` need `thrift compiler`, which is not present in CH.  Then we need to compile `thrift compiler`, which depends on `flex` and `bison`, these two are also not provided currently in CH. 

My question is how to introduce `thrift compiler`, `flex` and `bison` in CH. Currently I think there are 3 workarounds:
1. Generate *.cpp *.h files by hive_metastore.thrift advancely. Then we are able to compile Hive Engine without `thrift compiler`
2. Modify dockerfile correspond to building environment, install `flex` and `bison` advancely. Compile `thrift compiler` in cmake.
3. Add `contrib/bison` and `contrib/flex` in CH, and build them all from source code. I'am afraid it is not a easy work. 

Which way do you advise to go? @kssenii

Reply:
Lets try the first option.

Reply:
Ok, thanks for you advice.